### ISSUE TITLE
chai-json-schema needs a new owner

### ISSUE THREAD

**Issue by Bartvds** [chaijs/chai-json-schema#23]
I've completely crashed out of the node loop so this module desperately needs a new owner. 

If anyone wants to adopt please leave a message and we can arrange ownership transfer of repos and package as per npm guidelines.

**Comment by Petah** [comment]
Sad :(

**Comment by JrSchild** [comment]
Hi @Bartvds,
I am a developer at Enrise and chai-jason-schema is used in many of our projects. Today we discussed this with the team and we are open to adopting and maintaining the project.

Let me know if you are open to it as well.

Regards,
Joram

**Comment by Bartvds** [comment]
@JrSchild Sure, adopting by a company would be great.

Note: another of my Chai module's (`chai-fs`) was recently adopted and moved into the Chai.js Github org. I haven't asked for them to adopt this one as well since this is a bit more exotic and needs a bit of json-schema knowledge which I wouldn't dump on the organization without someone to pick it up.

I can ping the org manager who moved chai-fs for me; I'm sure we can arrange for you to become core maintainer with all the access rights (they make github teams for adopted modules and are cool with npm co-ownership).

If you like to continue work on chai-json-schema but with a solid long-term home in the org then maybe this would be an ideal option? This way we keep it in the community while giving you guys the power to move it forward.

**Comment by JrSchild** [comment]
Hi Bart,

We think it is a great idea to host the project at the Chai organization. Ensuring the project's long term support, together with our commitment, we think this would be a very good fit. As Enrise continues to grow, we slowly want to pick these kind of projects up to give something back to the community. Not only node.js related but also on other fronts as we have multidisciplinary teams.

It would be nice if you could ask the organization to host project there. I'm looking out to hear from you.

**Comment by Bartvds** [comment]
Cool, reviving some lost modules is a helpful way to get into open source.

Let's ping @keithamus

tl;dr: @JrSchild and company offer to maintain one of my neglected Chai modules and we think we could/should move this into the Chai.js org as well (and assign them a team and npm ownership). What do you think?

**Comment by keithamus** [comment]
Hey @Bartvds and @JrSchild. This all sounds good to me 😄. I'm happy to move this module into the chaijs main org, and set up both of you as maintainers, and also as team admins (if that's how you'd like it to be) so you can have full autonomy and add more maintainers as you see fit.

**Comment by JrSchild** [comment]
Thank you @keithamus, that is good news. Could you two arrange the transfer? Regarding the team admins, it is up to you, @Bartvds, if you would like to stick around. You're more than welcome of course.

**Comment by Bartvds** [comment]
@keithamus Very cool. 

How you want to do this? IIRC from last time it'll work smoothest if I transfer this repo to your account so you can move it in into the org (assuming you don't have it forked :)

**Comment by keithamus** [comment]
@Bartvds just renamed by fork, so you should be able to transfer it cleanly. It definitely worked smoothest when you transferred to me first. LMK when you do and I can move it to chaijs org 😄

**Comment by Bartvds** [comment]
@keithamus I tried transfering the repos to you but I get that "keithamus already has a repository in the Bartvds/chai-json-schema network" message. 

When we moved `chai-fs` a while ago we had same issue and only resolved after you deleted your renamed fork. :worried:

**Comment by keithamus** [comment]
I thought before we just renamed it. I've deleted it now, feel free to transfer.

**Comment by Bartvds** [comment]
@keithamus Cool, the request is in the pipe, you should receive a notification soon.

@JrSchild Could you let me know your npm user name so I can grant you ownership when the transfer is complete?

**Comment by JrSchild** [comment]
That's good news! We will be using our company username: 'enrise'.

**Comment by keithamus** [comment]
The way we typically manage releases in the chaijs org, is that we have a bot automate the releases using travis - this way the releases happen on a clean and manageable environment. We should probably set up the same automation for this repo also.

**Comment by keithamus** [comment]
Okay, permissions have all been set up in the org. @JrSchild and @PaulThompson you both have invites to join the `chai-json-schema` team on the chaijs org.

**Comment by PaulThompson** [comment]
Thanks.
Also in terms of npm ownership, There's also https://www.npmjs.com/package/chai2-json-schema and 
https://www.npmjs.com/package/chai3-json-schema variants by @darrrk 

It would be great to bring those in, in a compatible way.

Updating of those chai and tv4 dependencies has been the main source of interest and pull-requests against chai-json-schema.

**Comment by JrSchild** [comment]
Thank you @keithamus. Could you give me extra rights so I can add some of my coworkers to the team? That would be useful for PR reviewing/merging.

@PaulThompson I agree, it seems confusing to have multiple published modules under different names. As far as I can tell, chai2-json-schema is exactly identical to chai-json-schema, just published under a different name. The only difference with chai3-json-schema is the updated chai dependency.
When bringing the patches into this project, they will become obsolete. The obvious choice is this one, most stars/downloads and hosted under the chaijs organization.

**Comment by JrSchild** [comment]
Hi @keithamus I have sent you an email which I got from the commit log history. I'm not sure if it's the right one. We would like to set our internal team up in the project but I have insufficient permission. Is there anything you can do about that?

After that is done, this ticket can finally be closed.

**Comment by keithamus** [comment]
@Bartvds could you please add `chaijs` as a user on the npm package (https://www.npmjs.com/package/chai-json-schema/access)

**Comment by Bartvds** [comment]
@keithamus Yes of course. I'm at the office now but will do when I get back to my home computer tonight.

**Comment by JrSchild** [comment]
To round things up.

The project is now hosted under the chaijs organization. My company [Enrise](https://enrise.com/) volunteered to maintain it. Chai-json-schema is probably pretty feature complete as it is. We will be doing the necessary maintenance to make sure it doesn't break and to make sure it stays compatible with the recent chai versions.
Internally it is a team effort to do PR-reviews, merges, releases and handle feature requests if/when they come in. All discussion will of course be open and on this GitHub project. I have added a few of my colleagues as collaborators.

The short term goal is to do a release with chai 3.0+ support. Furthermore we would like to even things out as much as possible with the main chaijs project for consistency and transparency. When the npm user is added we can push releases by simply tagging the master branch.

**Comment by Bartvds** [comment]
@keithamus Yesterday got messy but I just added the new npm owners.

@JrSchild It's your party now, I'm not really active with node anymore so you're free to maintain and develop as you think is right. Consistency seems a good goal though; I'm not sure how consistent I was (probably not very :wink:).
